lms/djangoapps/verify_student/houston_encrypt.py

```diff
@@ -0,0 +1,125 @@
+    generate_signed_message
+)
+
+def create_houston_request(receipt_id, photo_id_key, copy_id_photo_from=None):
+    """
+    Construct the HTTP request to the photo verification service.
+    Keyword Arguments:
+        copy_id_photo_from (SoftwareSecurePhotoVerification): If provided, re-send the ID photo
+            data from this attempt.  This is used for reverification, in which new face photos
+            are sent with previously-submitted ID photos.
+    Returns:
+        tuple of (header, body), where both `header` and `body` are dictionaries.
+    """
+    scheme = "https" if settings.HTTPS == "on" else "http"
+    callback_url = "{}://{}{}".format(
+        scheme, settings.SITE_NAME, reverse('verify_student_results_callback')
+    )
+    # If we're copying the photo ID image from a previous verification attempt,
+    # then we need to send the old image data with the correct image key.
+    config = settings.VERIFY_STUDENT["SOFTWARE_SECURE"]
+    bucket = config["S3_BUCKET"]
+    s3Client = boto3.client('s3', 
+        aws_access_key_id=config["AWS_ACCESS_KEY"], 
+        aws_secret_access_key=config["AWS_SECRET_KEY"], 
+        region_name="us-east-1"
+    )
+    presigned_url_photo = s3Client.generate_presigned_url(
+        'get_object',
+        Params = {'Bucket': bucket, 'Key': 'face/{}'.format(receipt_id)},
+        ExpiresIn = 600
+    )
+    presigned_url_id = s3Client.generate_presigned_url(
+        'get_object',
+        Params = {'Bucket': bucket, 'Key': 'photo_id/{}'.format(receipt_id)},
+        ExpiresIn = 600
+    )
+    headers = {
+        "Content-Type": "application/json",
+        "Date": formatdate(timeval=None, localtime=False, usegmt=True)
+    }
+    houston_project_id = settings.VERIFY_STUDENT["HOUSTON_STU"]["HOUSTON_PROJECT_ID"]
+    houston_organisation_id = settings.VERIFY_STUDENT["HOUSTON_STU"]["HOUSTON_ORGANIZATION_ID"]
+
+    access_key = settings.VERIFY_STUDENT["HOUSTON_STU"]["API_ACCESS_KEY"]
+    secret_key = settings.VERIFY_STUDENT["HOUSTON_STU"]["API_SECRET_KEY"]
+    
+    body_for_signature = {"EdX-ID": str(receipt_id)}
+    _message, _sig, authorization = generate_signed_message(
+        "POST", headers, body_for_signature, access_key, secret_key
+    )
+
+    headers['Authorization'] = authorization
+    body = {
+        "organisationId": houston_organisation_id,
+        "projectId": houston_project_id,
+        "preSignedUrlId": presigned_url_id,
+        "preSignedUrlPhoto": presigned_url_photo,
+        "photoId": str(receipt_id),
+        "username": "tono",
+        "email": "tono@tono.com", 
+        "callbackUrl": callback_url,
+        "photoIdKey": photo_id_key
+    }
+    
+    return headers, body
+    
+def send_houston_request(receipt_id, photo_id_key, copy_id_photo_from=None):
+    """
+    Assembles a submission to Houston Stu and sends it via HTTPS.
+    Keyword Arguments:
+        copy_id_photo_from (SoftwareSecurePhotoVerification): If provided, re-send the ID photo
```

Comment:
again SoftwareSecurePhotoVerification?

Reply:
Because it uses the SoftwareSecureVerification model and it's fields.